pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -305,6 +307,7 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 		}
 
 		if managementCluster != nil {
+			// TODO: Does this belong here or in the manager?
```

Comment:
Both the apiserver and the manager have use for this secret. The apiserver allows users to create a managed cluster and obtain an installation manifests. I believe since Manager has a dependency on the ApiServer, it would make sense to create it here to prevent another rollout of this deployment.

Reply:
In multi-tenancy, this is a per-tenant secret - it's not just one per management cluster, and so it needs to be done in the manager controller which is what led me to this question.

I think it's OK to restart this pod if needed, though obviously would be nice to avoid it if possible. Either way, I think some cleanup can be done here in a future PR even if we keep provisioning it in two places